src/main/java/com/github/tomakehurst/wiremock/http/AbstractRequestHandler.java

```diff
@@ -102,6 +103,8 @@ public void handle(Request request, HttpResponder httpResponder, ServeEvent orig
 
     serveEvent.afterSend();
     afterResponseSent(serveEvent, response);
+
+    RequestCache.onRequestEnd();
```

Comment:
There is the risk that if, for any reason, an exception is thrown, the cache never gets to be cleaned.